Markdown 3.4.1 breaks converters/_tests/test_markdown_in_out.py

With the today released Markdown version 3.4.1 the test_markdown_in_out.py fails. The first error is:

```
src/moin/converters/_tests/test_markdown_in_out.py:131: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <moin.converters._tests.test_markdown_in_out.TestConverter object at 0x7fdd3b8614c0>, input = '|Tables|Are|Very|Cool|\n|------|:----:|-----:|:-----|\n|col 2 is|centered|$12|Gloves|\n'
output = '|Tables|Are|Very|Cool|\n|------|:----:|-----:|:-----|\n|col 2 is|centered{: class="center"}|$12{: class="right"}|Gloves{: class="left"}|\n', args = {}, skip = None

    def do(self, input, output, args={}, skip=None):
        if skip:
            pytest.skip(skip)
        out = self.conv_in(input, 'text/x-markdown;charset=utf-8', **args)
        out = self.conv_out(self.handle_input(self.serialize_strip(out)), **args)
        # assert self.handle_output(out) == output
>       assert self.handle_output(out).strip() == output.strip()  # TODO: revert to above when number of \n between blocks in moinwiki_out.py is stable
E       assert '|Tables|Are|...ign: left;"}|' == '|Tables|Are|...lass="left"}|'
E           |Tables|Are|Very|Cool|
E         - |------|:----:|-----:|:-----|
E         ?         -    ^      ^ ^
E         + |------|------|------|------|
E         ?             ^^      ^ ^
E         - |col 2 is|centered{: class="center"}|$12{: class="right"}|Gloves{: class="left"}|
E         + |col 2 is|centered{: style="text-align: center;"}|$12{: style="text-align: right;"}|Gloves{: style="text-align: left;"}|

```

Obviously the generated html code changed from e.g. `class="center"` to `style="text-align: center;"`

I appreciate help

--- a/setup.py
+++ b/setup.py
@@ -74,7 +74,7 @@ setup_args = dict(
         'Babel==2.9.1',  # Babel 2.10.1 breaks quickinstall.py #1239
         'blinker>=1.1',  # event signalling (e.g. for change notification trigger)
         'docutils<0.18',  # reST markup processing, see #1141; sphinx 4.3.0 requires docutils<0.18,>=0.14
-        'Markdown<3.4.0',  # Markdown markup processing TODO see #1290
+        'Markdown>=3.4.1',  # Markdown markup processing
         'Flask<2.0.0',  # micro framework
         'Flask-Babel>=0.11.1',  # i18n support
         'Flask-Caching>=1.2.0',  # caching support

--- a/src/moin/converters/_tests/test_markdown_in_out.py
+++ b/src/moin/converters/_tests/test_markdown_in_out.py
@@ -94,9 +94,8 @@ class TestConverter:
     data = [
         ('[TOC]\n',
          '[TOC]\n'),
-        # TODO: result changes with Markdown 3.4.1, see #1290
         ('|Tables|Are|Very|Cool|\n|------|:----:|-----:|:-----|\n|col 2 is|centered|$12|Gloves|\n',
-         '|Tables|Are|Very|Cool|\n|------|:----:|-----:|:-----|\n|col 2 is|centered{: class="center"}|$12{: class="right"}|Gloves{: class="left"}|\n'),
+         '|Tables|Are|Very|Cool|\n|------|:----:|-----:|:-----|\n|col 2 is|centered|$12|Gloves|\n'),
         # TODO: wrong output, creates indented blockcode, loses fenced code language
         # fix probably requires replacing site-packages/markdown/extensions/codehilite.py
         ('``` javascript\nvar s = "JavaScript syntax highlighting";\nalert(s);\n```\n',
@@ -175,11 +174,11 @@ class TestConverter:
         ('|A|B|C|\n|-|-|-|\n|1|2|3|\n',
          '|A|B|C|\n|------|------|------|\n|1|2|3|\n'),
         ('|A|B|C|\n|:-|:-:|-:|\n|1|2|3|\n',
-         '|A|B|C|\n|:-----|:----:|-----:|\n|1{: class="left"}|2{: class="center"}|3{: class="right"}|\n'),
+         '|A|B|C|\n|:-----|:----:|-----:|\n|1|2|3|\n'),
         ('A|B|C\n-|-|-\n1|2|3\n',
          '|A|B|C|\n|------|------|------|\n|1|2|3|\n'),
         ('`A`|*B*|_C_\n:-|:-:|-:\n1|2|3\n',
-         '|`A`|*B*|*C*|\n|:-----|:----:|-----:|\n|1{: class="left"}|2{: class="center"}|3{: class="right"}|\n'),
+         '|`A`|*B*|*C*|\n|:-----|:----:|-----:|\n|1|2|3|\n'),
     ]
 
     @pytest.mark.parametrize('input,output', data)

--- a/src/moin/converters/markdown_in.py
+++ b/src/moin/converters/markdown_in.py
@@ -356,11 +356,11 @@ class Converter:
 
     def visit_th(self, element):
         attrib = self.verify_align_style(element.attrib)
-        return self.new_copy(html.th, element, attrib=attrib)
+        return self.new_copy(moin_page.table_cell_head, element, attrib=attrib)
 
     def visit_td(self, element):
         attrib = self.verify_align_style(element.attrib)
-        return self.new_copy(html.td, element, attrib=attrib)
+        return self.new_copy(moin_page.table_cell, element, attrib=attrib)
 
     def visit(self, element):
         # Our element can be converted directly, just by changing the namespace

--- a/src/moin/converters/markdown_in.py
+++ b/src/moin/converters/markdown_in.py
@@ -35,7 +35,6 @@ from moin.utils.mime import Type, type_moin_document
 from moin import log
 logging = log.getLogger(__name__)
 
-
 html_in_converter = HTML_IN_Converter()
 block_elements = 'p h blockcode ol ul pre address blockquote dl div fieldset form hr noscript table'.split()
 BLOCK_ELEMENTS = {moin_page(x) for x in block_elements}
@@ -348,19 +347,19 @@ class Converter:
             return href
         return self.new_copy(moin_page.a, element, attrib)
 
-    def convert_align_to_class(self, attrib):
-        attr = {}
-        alignment = attrib.get('align')
-        if alignment in ('right', 'center', 'left'):
-            attr[moin_page.class_] = alignment
-        return attr
+    def verify_align_style(self, attrib):
+        alignment = attrib.get('style')
+        if alignment and alignment in ('text-align: right;', 'text-align: center;', 'text-align: left;'):
+            attrib = {moin_page.style: attrib.get('style')}
+            return attrib
+        return {}
 
     def visit_th(self, element):
-        attrib = self.convert_align_to_class(element.attrib)
+        attrib = self.verify_align_style(element.attrib)
         return self.new_copy(html.th, element, attrib=attrib)
 
     def visit_td(self, element):
-        attrib = self.convert_align_to_class(element.attrib)
+        attrib = self.verify_align_style(element.attrib)
         return self.new_copy(html.td, element, attrib=attrib)
 
     def visit(self, element):

--- a/src/moin/converters/markdown_out.py
+++ b/src/moin/converters/markdown_out.py
@@ -160,7 +160,7 @@ class Converter:
         if f:
             ret = f(elem)
             if elem.tag.name not in ('separator', 'blockcode', 'code', 'div', 'big', 'small', 'sup', 'sub', 'th',
-                                     'emphasis', 's', 'ins', 'u', 'span', ):
+                                     'emphasis', 's', 'ins', 'u', 'span', 'table', ):
                 attrib = self.attribute_list(elem)
                 if attrib:
                     if ret.endswith('#\n'):
@@ -456,6 +456,7 @@ class Converter:
         self.status.pop()
         # markdown tables must have headings
         if '----' not in ret:
+            # style: text-align gets lost here
             rows = ret.split('\n')
             header = rows[0][1:-1]  # remove leading and trailing |
             cells = header.split('|')

--- a/src/moin/converters/markdown_out.py
+++ b/src/moin/converters/markdown_out.py
@@ -126,10 +126,6 @@ class Converter:
                         ret = '\n'
                 if child == '\n' and getattr(elem, 'level', 0):
                     child = child + ' ' * (len(''.join(self.list_item_labels[:-1])) + len(self.list_item_labels[:-1]))
-                if elem.tag.name == 'td':
-                    attrib = self.attribute_list(elem)
-                    if attrib:
-                        child += attrib
                 childrens_output.append('{0}{1}'.format(ret, child))
                 self.last_closed = 'text'
         out = join_char.join(childrens_output)
@@ -472,11 +468,11 @@ class Converter:
         # used for reST to moinwiki conversion, maybe others that generate table head
         separator = []
         for th in elem[0]:
-            if th.attrib.get(moin_page.class_, None) == 'center':
+            if th.attrib.get(moin_page.style, None) == 'text-align: center;':
                 separator.append(':----:')
-            elif th.attrib.get(moin_page.class_, None) == 'left':
+            elif th.attrib.get(moin_page.style, None) == 'text-align: left;':
                 separator.append(':-----')
-            elif th.attrib.get(moin_page.class_, None) == 'right':
+            elif th.attrib.get(moin_page.style, None) == 'text-align: right;':
                 separator.append('-----:')
             else:
                 separator.append('------')

--- a/src/moin/converters/moinwiki_out.py
+++ b/src/moin/converters/moinwiki_out.py
@@ -625,8 +625,14 @@ class Converter:
         self.table_rowclass = ''
         return ret + Moinwiki.table_marker + '\n'
 
+    def open_moinpage_th(self, elem):
+        return self.open_moinpage_table_cell_head(self, elem)
+
     def open_moinpage_table_cell_head(self, elem):
         return self.open_moinpage_table_cell(elem)
+
+    def open_moinpage_td(self, elem):
+        return self.open_moinpage_table_cell(self, elem)
 
     def open_moinpage_table_cell(self, elem):
         table_cellclass = elem.attrib.get(moin_page.class_, '')

